core/decorations.py

```diff
@@ -24,10 +24,10 @@ def handle_publish_decorations(session: Session, decorations_params: Any) -> Non
 
     for sv in session_buffer.session_views:
         try:
-            phantom_set = getattr(sv, "_lsp_metals_decorations")
+            phantom_set = getattr(session_buffer, field_name)
         except AttributeError:
             phantom_set = sublime.PhantomSet(sv.view, phantom_key)
-            setattr(sv, "_lsp_metals_decorations", phantom_set)
+            setattr(session_buffer, field_name, phantom_set)
```

Comment:
@rchl so far the feature works but is this what you mean ?

Reply:
Remove the `for sv in session_buffer.session_views:` loop also since you only need to create and store phantoms once on `session_buffer`.

> BTW LSP-dart is also storing the phantom set in the view

Doesn't mean that it's correct. :)

Reply:
> Remove the for sv in session_buffer.session_views: loop also since you only need to create and store phantoms once on session_buffer.

But a view is needed to create a phantom set: `sublime.PhantomSet(sv.view, phantom_key)`. how do I get a single one ?
Is there a guarentee that there would be only one attached to the session buffer?

Reply:
The fact that it needs a view is a bit of a weird design on the ST API side IMO. It routes things to the Buffer anyway.

There is always guarantee that there is at least one view attached to the SessionBuffer and you can just get the first one.

Reply:
I used `view = list(session_buffer.session_views)[0].view` to get first element of weakset as It doesn't implement iterator so `next(set)` doesn't work. Let me know if there is a better way.
Or is it safe to use the view field ? https://github.com/sublimelsp/LSP/blob/main/plugin/session_buffer.py#L67

Reply:
Could do `next(iter(self.session_views)).view` but converting to list sounds fine to me too.

To be I'm not sure if it's 100% guaranteed that there will always be a view. Might want to put that inside the try block.

> Or is it safe to use the view field ?

Not really supposed to since it's not on a public `SessionBufferProtocol` interface.